app/src/main/java/org/schabi/newpipe/local/feed/FeedViewModel.kt

```diff
@@ -54,22 +60,24 @@ class FeedViewModel(
             FeedEventManager.events(),
             toggleShowPlayedItemsFlowable,
             toggleShowFutureItemsFlowable,
+            toggleShowShortsFlowable,
             feedDatabaseManager.notLoadedCount(groupId),
             feedDatabaseManager.oldestSubscriptionUpdate(groupId),
 
-            Function5 { t1: FeedEventManager.Event, t2: Boolean, t3: Boolean,
-                t4: Long, t5: List<OffsetDateTime> ->
-                return@Function5 CombineResultEventHolder(t1, t2, t3, t4, t5.firstOrNull())
+            Function6 { t1: FeedEventManager.Event, t2: Boolean, t3: Boolean, t4: Boolean,
+                t5: Long, t6: List<OffsetDateTime> ->
+                return@Function6 CombineResultEventHolder(t1, t2, t3, t4, t5, t6.firstOrNull())
             }
         )
         .throttleLatest(DEFAULT_THROTTLE_TIMEOUT, TimeUnit.MILLISECONDS)
         .subscribeOn(Schedulers.io())
         .observeOn(Schedulers.io())
-        .map { (event, showPlayedItems, showFutureItems, notLoadedCount, oldestUpdate) ->
+        .map { (event, showPlayedItems, showFutureItems, showShorts, notLoadedCount, oldestUpdate) ->
             val streamItems = if (event is SuccessResultEvent || event is IdleEvent)
                 feedDatabaseManager
                     .getStreams(groupId, showPlayedItems, showFutureItems)
                     .blockingGet(arrayListOf())
+                    .filter { s -> showShorts || s.stream.duration > 0 }
```

Comment:
Please explain this filter. Do shorts have an unknown duration (i.e. -1))? 

What about other services? When there is an unknown duration for e.g. a media.ccc.de or PeerTube stream, they are removed by your filter, too. I suggest to only remove streams with the YouTube service id.